sub-packages/bionemo-dnadl/src/bionemo/dnadl/io/dataset.py

```diff
@@ -0,0 +1,189 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-NvidiaProprietary
+#
+# NVIDIA CORPORATION, its affiliates and licensors retain all intellectual
+# property and proprietary rights in and to this material, related
+# documentation and any modifications thereto. Any use, reproduction,
+# disclosure or distribution of this material and related documentation
+# without an express license agreement from NVIDIA CORPORATION or
+# its affiliates is strictly prohibited.
+
+import os
+from abc import ABC
+from pathlib import Path
+from typing import Any, Callable, List, Optional
+
+import polars as pl
+import pysam
+import torch
+from torch.utils.data import Dataset
+
+from bionemo.dnadl.tools.genome_editing import create_personal_sequence
+from bionemo.dnadl.tools.genome_interval import GenomeInterval
+from bionemo.dnadl.tools.vcf import SampleId, read_variants_in_interval
+from bionemo.noodles.nvfaidx import NvFaidx
+
+
+DNATokenizer = Callable[[str], torch.Tensor]
+
+
+class Genome:
+    """Class that creates a genome object from a fasta file. It can be used to extract sequences from the genome."""
+
+    def __init__(self, fasta_file: str | os.PathLike):
+        """Instantiate the class."""
+        fasta_file = Path(fasta_file)
+        assert fasta_file.exists(), "path to fasta file must exist"
+        self.seqs = NvFaidx(str(fasta_file))
+
+    def adjust_interval(self, interval: GenomeInterval, context_length: int) -> GenomeInterval:
+        """Used to extend an interval by a fixed amount on both sides."""
+        start, end = interval.start, interval.end
+        interval_length = interval.end - interval.start
+        chromosome_length = self.seqs[interval.chromosome].length
+        if interval_length >= context_length:
+            return interval
+
```

Comment:
This docstring really needs help. I arrived here from the method below:

its clear context length expands the interval, and is truncated if it reaches the max. What are the bounds of sequence length that this can return? it seems like it could return any length region, which is okay, but would be best to be clear about that in the docstring.